src/commands/protocols/finality/receiver/publish-finality-save-ack-command.js

```diff
@@ -20,7 +20,13 @@ class PublishFinalitySaveAckCommand extends Command {
      * @param command
      */
     async execute(command) {
-        const { ual, publishOperationId, blockchain, operationId, remotePeerId } = command.data;
+        const { ual, publishOperationId, blockchain, operationId, remotePeerId, state } =
+            command.data;
+
+        let ualWithState = ual;
+        if (state) {
+            ualWithState = `${ual}:${state}`;
```

Comment:
It would be best to define functions for UAL manipulation defined in ualService. It would make it less error prone and easier to maintain.

Reply:
Make sense we will fix it in different pr